sensing/pointcloud_preprocessor/src/outlier_filter/ring_outlier_filter_nodelet.cpp

```diff
@@ -44,82 +44,116 @@ RingOutlierFilterComponent::RingOutlierFilterComponent(const rclcpp::NodeOptions
     std::bind(&RingOutlierFilterComponent::paramCallback, this, _1));
 }
 
-void RingOutlierFilterComponent::filter(
+// TODO(sykwer): Temporary Implementation: Rename this function to `filter()` when all the filter
+// nodes conform to new API. Then delete the old `filter()` defined below.
+void RingOutlierFilterComponent::faster_filter(
   const PointCloud2ConstPtr & input, [[maybe_unused]] const IndicesPtr & indices,
-  PointCloud2 & output)
+  PointCloud2 & output, const TransformInfo & transform_info)
 {
+  (void)transform_info;
+
   std::scoped_lock lock(mutex_);
   stop_watch_ptr_->toc("processing_time", true);
-  std::unordered_map<uint16_t, std::vector<std::size_t>> input_ring_map;
-  input_ring_map.reserve(128);
-  sensor_msgs::msg::PointCloud2::SharedPtr input_ptr =
-    std::make_shared<sensor_msgs::msg::PointCloud2>(*input);
+
+  output.point_step = sizeof(PointXYZI);
+  output.data.resize(output.point_step * input->width);
+  size_t output_size = 0;
 
   const auto ring_offset =
     input->fields.at(static_cast<size_t>(autoware_point_types::PointIndex::Ring)).offset;
-  for (std::size_t idx = 0U; idx < input_ptr->data.size(); idx += input_ptr->point_step) {
-    input_ring_map[*reinterpret_cast<uint16_t *>(&input_ptr->data[idx + ring_offset])].push_back(
-      idx);
-  }
-
-  PointCloud2Modifier<PointXYZI> output_modifier{output, input->header.frame_id};
-  output_modifier.reserve(input->width);
-
-  std::vector<std::size_t> tmp_indices;
-  tmp_indices.reserve(input->width);
-
   const auto azimuth_offset =
     input->fields.at(static_cast<size_t>(autoware_point_types::PointIndex::Azimuth)).offset;
   const auto distance_offset =
     input->fields.at(static_cast<size_t>(autoware_point_types::PointIndex::Distance)).offset;
```

Comment:
Just matter of readability: I think you could define a local helper lambda `getDistance` to get of all the verbose reinterpret cast after, like this:
```c++
auto getDistance = [&](const size_t idz) { return *reinterpret_cast<const float *>(&input->data[idx + distance_offset]); }
```
And the same for azimut. It would also limit the risk to misindexing.